course/resources/rqm-ecologized.md

```diff
@@ -0,0 +1,227 @@
+- Note that there are 2 or 3 of these topics per week in the original RQM schedule. Time-wise, this means that a topic approximately equals one class (lecture or tutorial). There are 29 topics here and will be 24 classes taught, so we will need to include three per week in some cases or reduce this to from 29 to 24.
+
+## Draft of proposed schedule
+- An immediate conflict to solve is how to ensure that the students get started with their project early, but also have the necessary knowledge about data analyses, R, and the ecology models that Martin wants to incorporate.
+- As we are now incorporating some ecology classes into this schedule, I think it is important to make sure enough time is dedicated to teaching sound principles around the data analyse pipeline, as this is still the main focus of the course imo.
+- Another point of consideration is whether we should teach all the R stuff with a common data set, so that we can streamline it for teaching the principles, and then hand out student data sets after a few weeks.
+    - I am leaning towards a common data set for the first weeks, since this would make it faster and force students to go through these procedures at least twice, once with the common data set and once with their own, which is advantageous or knowledge retention. I fear classes would become choppy if every group has their own data set specific problems come up, that's better dealt with in tutorials with the group specific TAs.
+- I originally split these topics into lecture and tutorial, but I will just call it 1 and 2 instead, since both of them will hopefully be in a computer lab. The third item under each week is for suggested homework.
+- Do we want simple quizzes at the start of each lecture, for the first ~5 weeks? This could serve as an easy to grade item and make sure that the students put effort in at home. It will make it easier for the students to know where they are at the drop date, and we can then reduce the quiz frequency as the course moves into the project based phase.
+
+### Week 1
+1. Introduction to the course, intro to the ecology part, motivation behind more reproducible workflows, talk briefly about spreadsheet and get straight into how to use R.
+    - Since Martin mentioned we should expect no programming experience from these students, I want to introduce programming at the first lecture already, at least conceptually.
+    - We can do this based on a [jupyter notebook I wrote previously](https://github.com/joelostblom/resources/blob/master/from-spreadsheets-to-pandas/from-spreadsheets-to-pandas.ipynb) with analogies between spreadsheet software and programming languages. It's in Python, but if you think this is a good idea, we can adopt it for R. If not, we can do something else.
+2. Basic R programming, vectors, data frames, loops, conditions, etc.
```

Comment:
I'm just thinking that this might be especially relevant for the theoretical models Martin talked about, where things like population of a species at the current timestep requires the population from the previous timestep, in which case a loop is needed. Might actually be very important for this class to be explicit about when a loop is necessary (e.g. simulations with discrete timesteps that require the previous state or solving ordinary differential equations using Euler's method etc.) vs. when vectorizing is better (which I agree is more often the case).